### ISSUE TITLE
Bundle Dakota with Carolina and make it pip installable 

### ISSUE THREAD

**Issue by DanSava** [equinor/Carolina#27]
Discussed with @dotfloat  and both Carolina and Dakota seem to be open source and it seems to be possible to bundle Dakota with Carolina and upload it to pip.

**Comment by pinkwah** [comment]
I've spent yesterday playing around with this and I can attest to that it's theoretically possible. Dakota isn't difficult to compile (ignore their compilation guides -- they're not wrong, but they make it seem much more complicated than it is), but it takes about two hours to compile on my laptop. Not good for PRs.

I figured out that we can create a docker image based on `manylinux2010_x86_64` with both boost and dakota pre-installed. However, unlike PyBind11, Boost.Python isn't a header-only library, meaning that it needs to be pre-compiled for each Python we wish to support. I've yet to decide whether this image will need multiple installations of boost and dakota.

Seems like this is the only thing that precludes us from an easy solution, as building *carolina* and bundling it with dakota and boost is automatic. This results in a 30MB python wheel.
